### ISSUE TITLE
How can I selectively load assets ?

### ISSUE THREAD

**Issue by kalote** [balderdashy/sails#3025]
Hello,

First, I really appreciate your work :+1: 

I'm trying to load specific JS file based on the current controller name (e.g. if i'm in the User controller, load files in "/user/*.js") or another variable. I'm using the automatic loading of CSS & JS via:

```
<!--STYLES-->
<!--STYLES END-->
```
and

```
<!--SCRIPTS-->
<!--SCRIPTS END-->
```

How can I retrieve specific information (a variable from a controller, the controller name, the session cookie, ...) in the pipeline.js file to selectively load JS files ?

Thanks

**Comment by danywheeler** [comment]
I run into the same problems ? is there something like variable name to add specified script that I need ?

**Comment by danywheeler** [comment]
I think this is an issue should submit to sails-linker

**Comment by danywheeler** [comment]
I've found a solution below.
grunt.config.set('sails-linker', {
		devJs: {
			options: {
				startTag: '',
				endTag: '',
				fileTmpl: '<script src="%s"></script>',
				appRoot: '.tmp/public'
			},
			files: {
				'.tmp/public/**/*.html': require('../pipeline').jsFilesToInject,
				'views/**/*.html': require('../pipeline').jsFilesToInject,
				'views/**/*.ejs': require('../pipeline').jsFilesToInject
			}
		}

chang startTag and endTage to fit you needs!

**Comment by danywheeler** [comment]
#2971 is it the same ?

**Comment by sailsbot** [comment]
Thanks for posting, @kalote.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
